routes/user.js

```diff
@@ -6,95 +6,63 @@ const crypto = require("crypto");
 const express = require("express");
 const router = express.Router();
 
+const models = require("../db/models");
 const EmailUtils = require("../email-utils");
-const Subscribers = require("../subscribers");
 
-// We send verification emails to addresses that want to subscribe.
-// These addresses are temporarily stored here, mapped to the unique
-// string token that is used for verification.
-const gUnverifiedEmails = new Map();
+const ResponseCodes = Object.freeze({
+  InternalError: 999,
+  EmailNotProvided: 100,
+  EmailNotFound: 101,
+  TokenMismatch: 102,
+});
+
+router.post("/add", async (req, res) => {
+  const user = await models.User.create({ email: req.body.email });
+  const url = `${AppConstants.SERVER_URL}/user/verify?state=${user.verificationToken}&email=${user.email}`;
```

Comment:
Should we be URI-encoding the `user.verificationToken` and `user.email` here since they are query params?

Reply:
They are hex digest values, which should be safe for uris.